src/Patch-registry.md

```diff
@@ -0,0 +1,81 @@
+# RFL Patch Registry
+
+This is an attempt to aggregate Rust abstractions for different areas of the
+kernel that may not yet be available upstream, as well as some leaf drivers
+and other implementations.
+
+If you are working on something that requires a listed abstraction that is not
+yet in-tree, or if you don't see something you need, it is a good idea to get 
+in touch [on Zulip](https://rust-for-linux.zulipchat.com/) or
+[on the list](https://lore.kernel.org/rust-for-linux/) to verify the latest
+status. Quite a few items are nearly ready to go, but have not been merged to
+avoid adding dead code.
+
+Usual status options are:
+
+- `merged`: this patch has been merged to upstream and is usable in recent
+  versions of the kernel.
+- `review`: this patchset has been submitted to the mailing list and is in a
+  nearly usable state, pending review or other blockers.
+- `RFC`: this patchset is currently collecting feedback on the mailing list. It
+  is likely not too far off from being upstream ready, but may be blocked on
+  not having an in-tree usecase. (some patchsets that would otherwise be
+  `review` but do not have an in-tree usecase have been downgraded to `RFC`
+  even if not an `RFC patch`).
+- `testing`: this patchset is generally considered good for further
+  experimentation and may have RFL review, but does not have review by relevant
+   maintainers.
+- `experimental`: this patchset is a work in progress.
+- `abandoned`: not recently maintained
+
+Note that things change frequently and the information in this table  may not
+always be the latest.
+
+| Target Area | Status | Author | Git Link | Mailing List Link |
```

Comment:
Thanks for putting all the information together!

I wonder whether bullet points/lists are better than tables, at very least, we can group things belonging to the same subsystem together. For example:

## Filesystem API

### vfs
  * Depends on: folio
  ...
## MM
### folio
...
### pages
...

## Filesystem implementions
### tarfs
  * Depends on: vfs 
  ...
### puzzle fs
...


Thoughts?

Reply:
Interesting, I like the thought of having dependencies. Maybe it would be good to keep the main abstractions in a table for quick reading, but the drivers and implementations could be in a list format?

I'll play around with it a bit

Reply:
Yes, tables are better for a quick look (or a rough picture of everything). However, I think people that use the information would probably do a search with key words, so it doesn't matter. Maybe a table for each large subsystem (like MM, filesystems, networking, drm, security, etc.)? But anyway, I don't have any problem with the current format, we can always improve the format of the information later ;-)

Reply:
Up to you :) if you think it is ok to merge as-is, I can make a PR doing some updates. Maybe even just an extra column saying the subsystem then a second one saying exactly what it is could work